aws-event-bridge/src/test/scala/akka/stream/alpakka/aws/eventbridge/EventBridgePublishMockSpec.scala

```diff
@@ -0,0 +1,176 @@
+    val putRequestEntry = PutEventsRequestEntry.builder().detail("event-bridge-message").build()
+    val publishRequest = requestDetail("event-bridge-message")
+    val putResult = resultResponse("event-id")
+
+    when(eventBridgeClient.putEvents(meq(publishRequest))).thenReturn(CompletableFuture.completedFuture(putResult))
+
+    val (probe, future) =
+      TestSource.probe[PutEventsRequestEntry].via(EventBridgePublisher.flow()).toMat(Sink.seq)(Keep.both).run()
+    probe.sendNext(putRequestEntry).sendComplete()
+
+    Await.result(future, 1.second) mustBe putResult :: Nil
+    verify(eventBridgeClient, times(1)).putEvents(meq(publishRequest))
+  }
+
+  it should "publish multiple PutEventsRequestEntry messages to eventbridge" in {
+    val putResult = resultResponse("event-id")
+
+    when(eventBridgeClient.putEvents(any[PutEventsRequest]())).thenReturn(CompletableFuture.completedFuture(putResult))
+
+    val (probe, future) =
+      TestSource.probe[Seq[PutEventsRequestEntry]].via(EventBridgePublisher.flowSeq()).toMat(Sink.seq)(Keep.both).run()
+    probe
+      .sendNext(Seq(entryDetail("eb-message-1")))
+      .sendNext(Seq(entryDetail("eb-message-2")))
+      .sendNext(Seq(entryDetail("eb-message-3")))
+      .sendComplete()
+
+    Await.result(future, 1.second) mustBe putResult :: putResult :: putResult :: Nil
+
+    val expectedFirst = requestDetail("eb-message-1")
+    verify(eventBridgeClient, times(1)).putEvents(meq(expectedFirst))
+
+    val expectedSecond = requestDetail("eb-message-2")
+    verify(eventBridgeClient, times(1)).putEvents(meq(expectedSecond))
+
+    val expectedThird = requestDetail("eb-message-3")
+    verify(eventBridgeClient, times(1)).putEvents(meq(expectedThird))
+  }
+
+  it should "fail stage if EventBridgeAsyncClient request failed" in {
+    val publishRequest = requestDetail("eb-message")
+
+    val promise = new CompletableFuture[PutEventsResponse]()
+    promise.completeExceptionally(new RuntimeException("publish error"))
+
+    when(eventBridgeClient.putEvents(meq(publishRequest))).thenReturn(promise)
+
+    val (probe, future) =
+      TestSource.probe[Seq[PutEventsRequestEntry]].via(EventBridgePublisher.flowSeq()).toMat(Sink.seq)(Keep.both).run()
+    probe.sendNext(Seq(entryDetail("eb-message"))).sendComplete()
+
+    a[RuntimeException] should be thrownBy {
+      Await.result(future, 1.second)
+    }
+
+    verify(eventBridgeClient, times(1)).putEvents(meq(publishRequest))
+  }
+
+  it should "fail stage if upstream failure occurs" in {
+    case class MyCustomException(message: String) extends Exception(message)
+
+    val (probe, future) =
+      TestSource.probe[Seq[PutEventsRequestEntry]].via(EventBridgePublisher.flowSeq()).toMat(Sink.seq)(Keep.both).run()
+    probe.sendError(MyCustomException("upstream failure"))
+
+    a[MyCustomException] should be thrownBy {
+      Await.result(future, 1.second)
+    }
+
+    verify(eventBridgeClient, never()).putEvents(any[PutEventsRequest]())
+  }
+
```

Comment:
Maybe we could add a test that simulates an error returned by the service.  There are some example error responses in the docs.  https://docs.aws.amazon.com/eventbridge/latest/userguide/add-events-putevents.html